gitpod-io/gitpod
# gp rebuild with isolated debug workspace

## Description


This approach creates an isolated debug workspace within the existing workspace using the **`docker run`** command. To access the debug workspace, a user adds the prefix **`debug-`** to the original workspace ID and uses the new ID as the debug workspace ID. See [[1](https://deploy-preview-3290--gitpod-kumquat.netlify.app/docs/configure/workspaces/testing-changes))] to learn more about the usage.

Both the original and debug workspaces share the same workspace, Docker, supervisor, and IDE mounts, but have separate networks. The debug workspace runs an HTTP proxy and SSH server to enable access. The workspace proxy uses the **`debug-`** prefix to forward traffic to the corresponding workspace.

## Related Issue(s)

Related to https://github.com/gitpod-io/gitpod/issues/7671

## How to test


- Start a workspace and then follow [[1](https://deploy-preview-3290--gitpod-kumquat.netlify.app/docs/configure/workspaces/testing-changes)] to use `gp rebuild`
- It should work with VS Code Browser and Desktop (with latest version of extensions).
- JB changes are in for the launcher, but it is hard to test in preview evns. We will test in production and improve later.

## Release Notes

```release-note
Allow to test the workspace configuration changes within the current workspace with gp rebuild.
```

## Documentation


## Build Options:

      Experimental feature to run the build with GitHub Actions (and not in Werft).
      If enabled this will build `install/preview`
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
